### DISCUSSION TITLE
ERROR: Package 'open-webui' requires a different Python: 3.12.3 not in '<3.12.0a1,>=3.11'

### DISCUSSION THREAD

**Discussion by Morrigan-Ship** [open-webui/open-webui#2596] [General]
Platform: windows
Python: v3.12.3
pip: 24.0

command:
`pip install open-webui`

error:
```
Defaulting to user installation because normal site-packages is not writeable
ERROR: Ignored the following versions that require a different python version: 0.1.124 Requires-Python <3.12.0a1,>=3.11; 0.1.125 Requires-Python <3.12.0a1,>=3.11; 0.2.0.dev1 Requires-Python <3.12.0a1,>=3.11; 0.2.0.dev2 Requires-Python <3.12.0a1,>=3.11
ERROR: Could not find a version that satisfies the requirement open-webui (from versions: none)
ERROR: No matching distribution found for open-webui
```

command:
`pip install open_webui-0.1.125-py3-none-any.whl`

error:
`ERROR: Package 'open-webui' requires a different Python: 3.12.3 not in '<3.12.0a1,>=3.11'`

**Comment by justinh-rahb** [comment]
We only support Python 3.11, and recommend using a virtual environment tool such as Conda.

**Reply by Morrigan-Ship** [reply]
> We only support Python 3.11

but there is a problem.
in Python 3.11 toml cannot be installed on windows.

**Reply by SAMEH-MUSTAFA1** [reply]
> We only support Python 3.11, and recommend using a virtual environment tool such as Conda.

!!!The release you're looking at is Python 3.11.10, a security bugfix release for the legacy 3.11 series!!!
,kindly make it to support latest version of Py

**Comment by willbasky** [comment]
In arch I use `pipx install --fetch-missing-python --python 3.11 open-webui` command to install it.

**Reply by iamrohitanshu** [reply]
> In arch I am using `pipx install --fetch-missing-python --python 3.11 open-webui` command to install it.

It worked for me in Windows 10 as well. Thank you!

**Reply by JasonBunting** [reply]
> It worked for me in Windows 10 as well. Thank you!

Same - and thank you for saying so, was going a bit crazy trying to get this done.

**Reply by slemfisk** [reply]
omg it was that simple? I tried with pip install open-webui -- got error

**Reply by GreenyZA** [reply]
What a Champion!! Thank you!!

**Comment by tanpenggood** [comment]
It's a sad thing. 😭

```
pip install open-webui
ERROR: Could not find a version that satisfies the requirement open-webui (from versions: none)
ERROR: No matching distribution found for open-webui
```

> Windows 11
> Python 3.13.2
> pip 25.0

**Reply by vallamost** [reply]
Please support Python 3.13 devs, only supporting 3.11 is unfortunate.

**Reply by shadowdao** [reply]
@vallamost You can make a request, but lets keep it polite please.

**Comment by redbellyblack** [comment]
You may want to make it very clear in the documentation online that the latest update in ubuntu (ie python 3.13) will not work with out creating a virtual environment to install older versions of python. it gets very frustrating when your own documentation doesnt even mention it.
